### ISSUE TITLE
UP032 Failing to Detect `format` based on Length

### ISSUE THREAD

**Issue by max-muoto** [astral-sh/ruff#10235]
Running into a weird edge case where UP032 is not picking up the usage of formats.

Take this example, UP032 finds nothing wrong here:

```python
idx = 1
foo = (
    "This is a test, this is a test, this is a test, this is a test {}, {} test".format(
        idx + 1, len([])
    )
)
```

However, if I reduce the length, it gets flagged:

```python
idx = 1
z = "This is a test, this is a test, {}, {} test".format(
   idx + 1, len([])
)
```

Ruff Version: 0.3.0
Command: `ruff check .`.

**Comment by MichaReiser** [comment]
Thanks. Agree, that this is unexpected

The rule has a specific check to opt out of providing a fix when the fixed string exceeds the configured line length (because it could introduce new lint errors). 

https://github.com/astral-sh/ruff/blob/184241f99affa895c5e8c1a2e144587cfaa2339a/crates/ruff_linter/src/rules/pyupgrade/rules/f_strings.rs#L450-L459

We do this for a couple of rules but there seems to be no pattern: 
1. Some rules don't provide a fix if the "fixed" string exceeds the configured line width
1. Some rules don't create a diagnostic when the fixed expression exceeds the configured line width. 

I understand 1. but am surprised by 2. Let me see if I can figure out what the reasoning behind it is.

**Comment by max-muoto** [comment]
> Thanks. Agree, that this is unexpected
> 
> The rule has a specific check to opt out of providing a fix when the fixed string exceeds the configured line length (because it could introduce new lint errors).
> 
> https://github.com/astral-sh/ruff/blob/184241f99affa895c5e8c1a2e144587cfaa2339a/crates/ruff_linter/src/rules/pyupgrade/rules/f_strings.rs#L450-L459
> 
> We do this for a couple of rules but there seems to be no pattern:
> 
> 1. Some rules don't provide a fix if the "fixed" string exceeds the configured line width
> 2. Some rules don't create a diagnostic when the fixed expression exceeds the configured line width.
> 
> I understand 1. but am surprised by 2. Let me see if I can figure out what the reasoning behind it is.

Thank! It would be nice if there was some way (for formatter users) to still fix it, but then reformat the code to fix the now violating line, but not sure if this is doable considering how they're detached. Perhaps this would work better as a setting (i.e. fix violating rules even if it leads to a resulting line greater than the line limit).
